### DISCUSSION TITLE
Auto continue

### DISCUSSION THREAD

**Discussion by den47999** [xtekky/gpt4free#1843] [Q&A]
Where in the code is the Auto continue code written? In which file?
![Screenshot_1](https://github.com/xtekky/gpt4free/assets/118296790/fb1fe3c4-9b07-427c-95e0-aa56065efdf2)

**Comment by hlohaus** [comment]
Only OpenaiChat supports auto_continue.  In: OpenaiChat.py

**Comment by den47999** [comment]
can you specify the file where this code is located? or write it here?
![Screenshot_1](https://github.com/xtekky/gpt4free/assets/118296790/fd619206-bbfd-4b4d-b30a-9ef4467b6984)

**Reply by hlohaus** [reply]
The full path is:

g4f/Provider/needs_auth/OpenaiChat.py

**Comment by den47999** [comment]
Can you tell me where and how to get this data?
![Screenshot_1](https://github.com/xtekky/gpt4free/assets/118296790/78ba8795-f2aa-4e1e-93e8-ac455a3f57d8)

**Reply by hlohaus** [reply]
Hey, you need to fill in these fields with cookie values. Gemini needs more than one cookie, so you might as well create a cookie file. It's explained in the readme.

**Comment by hlohaus** [comment]
Please provide the named cookie value from the provider's website. You can obtain this value using a cookie manager or the developer tools in your browser.

**Comment by den47999** [comment]
Put it in and it doesn't work, I'm a simple user and I'm having a hard time figuring it out :(
![Screenshot_1](https://github.com/xtekky/gpt4free/assets/118296790/9ee007a2-34fd-4c79-b9f8-7d39f50105f4)

**Comment by den47999** [comment]
![Screenshot_2](https://github.com/xtekky/gpt4free/assets/118296790/a029034a-da9c-4d70-8c78-1d9eebf61a93)

**Reply by den47999** [reply]
Put it in and it doesn't work, I'm a simple user and I'm having a hard time figuring it out :(

**Reply by hlohaus** [reply]
Could you please clarify the specific issue you are encountering? Is it related to the generation of images?

**Reply by den47999** [reply]
I have two questions: I do not know what to put what cookies read to use bing? Before without cookies worked chat. now I have none of the providers does not respond gives different errors!

**Comment by den47999** [comment]
In giga chat I create a picture but do not know where it is saved I would like the picture to be saved to the desktop.
![Screenshot_1](https://github.com/xtekky/gpt4free/assets/118296790/ae27b63a-c458-4ff1-8f55-0f578e147c69)

**Reply by hlohaus** [reply]
Hey, I don't have access to GigaChat, so I'm not sure how the images are generated.
